src/core/stdc/stdarg.d

```diff
@@ -34,93 +81,75 @@ unittest
     assert((-9).alignUp!8 == -8);
 }
 
-version (X86)
+
+/**
+ * The argument pointer type.
+ */
+version (SysV_x64)
 {
-    /*********************
-     * The argument pointer type.
-     */
-    alias char* va_list;
+    alias va_list = core.internal.vararg.sysv_x64.va_list;
+    public import core.internal.vararg.sysv_x64 : __va_list, __va_list_tag;
+}
+else version (AAPCS32)
+{
+    alias va_list = __va_list;
 
-    /**********
-     * Initialize ap.
-     * For 32 bit code, parmn should be the last named parameter.
-     * For 64 bit code, parmn should be __va_argsave.
-     */
-    void va_start(T)(out va_list ap, ref T parmn)
+    // need std::__va_list for C++ mangling compatibility (AAPCS32 section 8.1.4)
+    extern (C++, std) struct __va_list
     {
-        ap = cast(va_list) ((cast(void*) &parmn) + T.sizeof.alignUp);
+        void* __ap;
     }
+}
+else version (AAPCS64)
+{
+    alias va_list = __va_list;
 
-    /************
-     * Retrieve and return the next value that is type T.
-     * Should use the other va_arg instead, as this won't work for 64 bit code.
-     */
-    T va_arg(T)(ref va_list ap)
+    // https://github.com/ARM-software/abi-aa/blob/master/aapcs64/aapcs64.rst#definition-of-va-list
+    extern (C++, std) struct __va_list
     {
-        T arg = *cast(T*) ap;
-        ap += T.sizeof.alignUp;
-        return arg;
+        void* __stack;
+        void* __gr_top;
+        void* __vr_top;
+        int __gr_offs;
+        int __vr_offs;
     }
+}
+else
+{
+    alias va_list = char*;
+}
 
-    /************
-     * Retrieve and return the next value that is type T.
-     * This is the preferred version.
-     */
-    void va_arg(T)(ref va_list ap, ref T parmn)
-    {
-        parmn = *cast(T*) ap;
-        ap += T.sizeof.alignUp;
-    }
 
-    /*************
-     * Retrieve and store through parmn the next value that is of TypeInfo ti.
-     * Used when the static type is not known.
-     */
-    void va_arg()(ref va_list ap, TypeInfo ti, void* parmn)
+/**
+ * Initialize ap.
+ * parmn should be the last named parameter;
+ * for DMD and non-Windows x86_64 targets, it should be __va_argsave.
+ */
+version (X86)
+{
+    void va_start(T)(out va_list ap, ref T parmn)
```

Comment:
Can the version condition not be put inside the function?

Reply:
It's an intrinsic for DMD+x86_64 and LDC.

Reply:
If it is an intrinsic, that still means that the body _should_ be ignored.

Reply:
It would appear that it is not an intrinsic on x86 dmd, that should be fixed, but it won't block this pr.